packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+///     child: MouseRegion(
+///       cursor: SystemMouseCursors.text,
+///       child: Container(
+///         width: 200,
+///         height: 100,
+///         decoration: BoxDecoration(
+///           color: Colors.blue,
+///           border: Border.all(color: Colors.yellow),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// Assigning regions with mouse cursors on platforms that do not support mouse
+/// cursors, or when there are no mouse connected, will result in no-op.
+///
+/// ## Related classes
+///
+/// The following classes are designed to be created and used by the framework,
+/// therefore should not be directly used by widgets.
+///
+/// [MouseTrackerCursorMixin] is a class that manages states, and dispatches
+/// specific operations based on general mouse device updates.
+///
+/// [MouseCursorController] implements low-level imperative control by directly
+/// talking to the platform.
+///
+/// See also:
+///
+///  * [MouseRegion], which is a common way of assigning a region with a
+///    [MouseCursor].
+///  * [MouseTracker], which determines the cursor that each device should show,
+///    and dispatches the changing callbacks.
+///  * [SystemMouseCursors], which enumerates supported system cursors.
+@immutable
+abstract class MouseCursor with Diagnosticable {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const MouseCursor();
+
+  /// A very short pretty description of the mouse cursor.
+  ///
+  /// The [debugDescription] shoule be a few words that can differentiate
+  /// instances of a class to make debug information more readable. For example,
+  /// a [SystemMouseCursor] class with description "drag" will be printed as
+  /// "SystemMouseCursor(drag)".
+  ///
+  /// The [debugDescription] should not be null, but can be an empty string,
+  /// which means the class is not configurable.
+  String get debugDescription;
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    final String debugDescription = this.debugDescription;
+    if (minLevel.index >= DiagnosticLevel.info.index && debugDescription != null)
+      return '$runtimeType($debugDescription)';
+    return super.toString(minLevel: minLevel);
+  }
+}
+
+/// An interface for mouse cursors that have all resources prepared and ready to
+/// be sent to the system.
+///
+/// Although [PreparedMouseCursor] adds no changes on top of [MouseCursor], this
+/// class is designed to prevent unprepared cursor types from methods that
+/// directly talk to the system.
+abstract class PreparedMouseCursor extends MouseCursor {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const PreparedMouseCursor();
+}
+
+/// A mouse cursor that does nothing when activated.
```

Comment:
Imagine there is a platform view, where the system handles pointer events and set cursors accordingly (e.g. it's a clickable region with a textfield at the center). 

1. The `PlatformView` widget has to set some cursor. If its `cursor` is null, and for example it's stacked onto another button, then hovering over to the button will change the cursor to the button's cursor (even though the button's not visible).
2. The `PlatformView` widget can't be set to any actual cursors, because then it might overwrite the value that the system wants to set.

By setting a region as `NoopMouseCursor`, Flutter will not do anything (i.e. keep the last cursor) when the mouse enters the region, thus avoiding overwriting the request of the platform view while absorbing the cursor hit test.

The reason I nam them differently is because `NoopMouseCursor` is the "action" (activating this cursor does nothing), while `uncontrolled` is the "state" (this region wants Flutter to not control cursors).

Suggestions are welcome.

Reply:
Maybe add some of what you say here to the docs. E.g. that its purpose is to block other regions below it from changing the cursor.

It's also a little confusing that this doc claims it does nothing and the doc for SystemMouseCursors.uncontrolled says it does something ("release control"). Maybe unify the messaging?

Reply:
Ok how about this:

NoopMouseCursor:
```dart
/// A mouse cursor that doesn't change the cursor when activated.
/// 
/// Although setting a region's cursor to [NoopMouseCursor] doesn't change the 
/// cursor, it blocks regions behind it from changing the cursor, in contrast to
/// setting the cursor to null. More information about the usage of this class
/// can be found at [SystemMouseCursors.uncontrolled].
/// 
/// To use this class, use [SystemMouseCursors.uncontrolled]. Directly
/// instantiating this class is not allowed.
```

SystemMouseCursors.uncontrolled:
```dart
  /// A special value that doesn't change cursor by itself, but make a region
  /// blocks other regions behind it from changing the cursor.
  /// 
  /// When a pointer enters a region with a cursor of [uncontrolled], the pointer
  /// retains its previous cursor and keeps so until it moves out of the region.
  /// Technically, this region absorb the mouse cursor hit test without changing
  /// the pointer's cursor.
  /// 
  /// This is useful in a region that displays a platform view, which let the
  /// operating system handle pointer events and change cursors accordingly. This
  /// requires that the region's cursor must not be any Flutter cursor, since
  /// that might overwrite the system request upon pointer entering; the cursor
  /// must not be null either, since that allows the widgets behind the region to
  /// change cursors.
```